OpenRA.Mods.Common/FileInsensitive.cs

```diff
@@ -0,0 +1,78 @@
+#region Copyright & License Information
+
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+
+#endregion
+
+using System;
+using System.IO;
+using System.Linq;
+
+namespace OpenRA.Mods.Common
+{
+	// Sometimes specific archives, filesystems, environments or disc copies differ in their file or path capitalization.
+	// This class builds up the path by checking every segment for the right capitalization found.
+	public static class FileInsensitive
```

Comment:
The approach here doesn't sit well with me. IMO it might make more sense to add a `public static string ResolveCaseInsensitivePath(string path)` after `ResolveAssemblyPath` in `OpenRA.FileSystem.FileSystem`, and then have the users resolve the path and call `File.*` themselves?